ecosystem/typescript/sdk/src/transactions/transaction_worker.ts

```diff
@@ -0,0 +1,169 @@
+import EventEmitter from "eventemitter3";
+import { AptosAccount } from "../account";
+import { PendingTransaction } from "../generated";
+import { AptosClient, Provider } from "../providers";
+import { TxnBuilderTypes } from "../transaction_builder";
+import { AccountSequenceNumber } from "./account_sequence_number";
+
+export class TransactionWorker extends EventEmitter {
+  readonly provider: Provider;
+
+  readonly account: AptosAccount;
+
+  // current account sequence number
+  readonly accountSequnceNumber: AccountSequenceNumber;
+
+  // process has started
+  started: boolean;
+
+  // process has stopped
+  stopped: boolean;
+
+  // transactions payloads waiting to be generated and signed
+  // TODO support entry function payload from ABI builder
+  transactionsQueue: Array<TxnBuilderTypes.TransactionPayload> = [];
+
+  // signed transactions waiting to be submitted
+  outstandingTransactions: Array<[Promise<PendingTransaction>, bigint]> = [];
+
+  // transactions that have been submitted to chain
+  processedTransactions: Array<[string, bigint, any]> = [];
+
+  constructor(provider: Provider, account: AptosAccount) {
+    super();
+    this.provider = provider;
+    this.account = account;
+    this.started = false;
+    this.stopped = false;
+    this.accountSequnceNumber = new AccountSequenceNumber(provider, account);
+  }
+
+  /**
+   * Gets the current account sequence number,
+   * generates the transaction with the account sequence number,
+   * adds the transaction to the outstanding transaction queue
+   * to be processed later.
+   */
+  async submitTransactions() {
+    try {
+      if (this.transactionsQueue.length === 0) return;
+      const sequenceNumber = await this.accountSequnceNumber.nextSequenceNumber();
+      if (sequenceNumber === null) return;
+      const transaction = await this.generateNextTransaction(this.account, sequenceNumber);
+      if (!transaction) return;
+      const pendingTransaction = this.provider.submitSignedBCSTransaction(transaction);
+      this.outstandingTransactions.push([pendingTransaction, sequenceNumber]);
+    } catch (error: any) {
+      throw new Error(error);
+    }
+  }
+
+  /**
+   * Reads the outstanding transaction queue and submits the transaction to chain.
+   *
+   * If the transaction has fulfilled, it pushes the transaction to the processed
+   * transactions queue and fires a transactionsFulfilled event.
+   *
+   * If the transaction has failed, it pushes the transaction to the processed
+   * transactions queue with the failure reason and fires a transactionsFailed event.
+   */
+  async processTransactions() {
+    try {
```

Comment:
why add try statements if we don't have any error handling? maybe add some todos of what you intend to do?